src/shared/autoMiner.ts

```diff
@@ -0,0 +1,102 @@
+import { BitcoinNode } from 'shared/types';
+import store from 'store';
+import { AutoMineMode } from '../store/models/network';
+
+export interface AutoMinerEvent {
+  remainingPercentage: number;
+  mode: AutoMineMode;
+  mining: boolean;
+}
+
+export type AutoMinerListener = (event: AutoMinerEvent) => void;
+
+export class AutoMiner {
```

Comment:
While this class is pretty simple to understand, I don't think it's the right architectural fit the way the app is structured. It would be better if this was a store model class with it's state being tracked by easypeasy/Redux. Here's a few reasons why:
1. It should't import `'store'` directly into any other module. The state and actions should only be accessed from the instance stored in the `<StoreProvider>` in the `App` component.  
2. It wouldn't need the pubsub interface to comminucate with the component. It would just need to update the store's state and Redux will handle flowing that updated state down to the React components that use it (ex: `AutoMineButton`), re-rendering when necessary.
3. Writing unit tests for this class would be a bit challenging right now because of the  direct import of the `store`.
4. A small note but this `shared` is meant for types & utils that are shared between the Electron 'main' process code that manages the browser windows (in `/electron/`) and the `rendered` process code (in `/src/`) which is the React/Redux app.

I suggest moving this implementation into the `/src/store/models/` dir and refactoring it to be an easypeasy store like the others.

Reply:
That makes sense, thank you! I believe the next iteration (pushed already) is more aligned with these guidelines (although I suspect further changes might be needed relating to how auto-mining is now integrated into the state store).

Looking forward to your inputs on the latest version 👍